[Windows] When using the assistant in the command line, when pressing Enter to accept the generation, the character `^U` is placed in front of the code

### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

~

### Environment

Zed: v0.162.0 (Zed Dev 181c3724aa99e67ebc5f8147dce8d6ee2f77276d)
OS: Windows 10.0.22631
Memory: 63.2 GiB
Architecture: x86_64
GPU: AMD Radeon RX 5600 XT || AMD proprietary driver || 24.10.27.01 (AMD proprietary shader compiler)

### If applicable, add mockups / screenshots to help explain present your vision of the feature


https://github.com/user-attachments/assets/4b5ad0a6-d70f-4f48-8419-c654c001b1f0



### If applicable, attach your Zed.log file to this issue.

<details><summary>Zed.log</summary>

<!-- Click below this line and paste or drag-and-drop your log-->
```
2024-11-12T18:49:23.4597687+03:00 [WARN] skipping diagnostics update, no worktree found for path "c:\\Users\\taima\\.cargo\\registry\\src\\index.crates.io-6f17d22bba15001f\\anyhow-1.0.89\\src\\macros.rs"
2024-11-12T18:49:23.6995108+03:00 [INFO] building git repository, `.git` path in the worktree: "\\\\?\\G:\\CodeProjects\\RustroverProjects\\other\\zed\\.git"
2024-11-12T18:49:23.7520509+03:00 [INFO] Initializing default prettier with plugins {}
2024-11-12T18:49:23.7521404+03:00 [INFO] Initializing default prettier with plugins {}
2024-11-12T18:49:23.7540694+03:00 [INFO] Initializing default prettier with plugins {}
2024-11-12T18:49:23.776048+03:00 [INFO] Language server with id 38 sent unhandled notification taplo/didChangeSchemaAssociation:
{
  "documentUri": "file:///G:/CodeProjects/RustroverProjects/other/zed/crates/zed/Cargo.toml",
  "schemaUri": "https://json.schemastore.org/cargo.json",
  "meta": {
    "name": "Cargo Manifest",
    "description": "Manifest for Cargo, the Rust package manager and build tool",
    "source": "catalog",
    "catalog_url": "https://www.schemastore.org/api/json/catalog.json"
  }
}
2024-11-12T18:49:23.7768642+03:00 [INFO] Language server with id 38 sent unhandled notification taplo/didChangeSchemaAssociation:
{
  "documentUri": "file:///G:/CodeProjects/RustroverProjects/other/zed/Cargo.toml",
  "schemaUri": "https://json.schemastore.org/cargo.json",
  "meta": {
    "name": "Cargo Manifest",
    "description": "Manifest for Cargo, the Rust package manager and build tool",
    "source": "catalog",
    "catalog_url": "https://www.schemastore.org/api/json/catalog.json"
  }
}
2024-11-12T18:49:23.7930356+03:00 [INFO] Language server with id 38 sent unhandled notification taplo/didChangeSchemaAssociation:
{
  "documentUri": "file:///G:/CodeProjects/RustroverProjects/other/zed/Cross.toml",
  "schemaUri": null,
  "meta": null
}
2024-11-12T18:49:24.125923+03:00 [ERROR] Error { kind: Generic("Input watch path is neither a file nor a directory."), paths: [] }
2024-11-12T18:49:24.5007607+03:00 [INFO] Language server with id 38 sent unhandled notification taplo/didChangeSchemaAssociation:
{
  "documentUri": "file:///G:/CodeProjects/RustroverProjects/other/zed/crates/zed/Cargo.toml",
  "schemaUri": "https://json.schemastore.org/cargo.json",
  "meta": {
    "name": "Cargo Manifest",
    "description": "Manifest for Cargo, the Rust package manager and build tool",
    "source": "catalog",
    "catalog_url": "https://www.schemastore.org/api/json/catalog.json"
  }
}
2024-11-12T18:49:24.5056428+03:00 [INFO] Language server with id 38 sent unhandled notification taplo/didChangeSchemaAssociation:
{
  "documentUri": "file:///G:/CodeProjects/RustroverProjects/other/zed/Cargo.toml",
  "schemaUri": "https://json.schemastore.org/cargo.json",
  "meta": {
    "name": "Cargo Manifest",
    "description": "Manifest for Cargo, the Rust package manager and build tool",
    "source": "catalog",
    "catalog_url": "https://www.schemastore.org/api/json/catalog.json"
  }
}
2024-11-12T18:49:24.5066973+03:00 [INFO] Language server with id 38 sent unhandled notification taplo/didChangeSchemaAssociation:
{
  "documentUri": "file:///G:/CodeProjects/RustroverProjects/other/zed/Cross.toml",
  "schemaUri": null,
  "meta": null
}
2024-11-12T18:49:26.16913+03:00 [WARN] Generic lsp request to rust-analyzer failed: content modified
2024-11-12T18:49:26.1804077+03:00 [ERROR] content modified
2024-11-12T18:49:26.5034891+03:00 [WARN] Language server rust-analyzer (id 37) status update: Failed to run build scripts of some packages.
2024-11-12T18:49:29.9869977+03:00 [WARN] Generic lsp request to rust-analyzer failed: content modified
2024-11-12T18:49:30.0219764+03:00 [ERROR] inlay hint update task for range failed: Error {
    context: "inlay hint fetch task",
    source: Error {
        context: "inlay hints LSP request",
        source: "content modified",
    },
}
2024-11-12T18:49:34.8696933+03:00 [WARN] unhandled message: TaskStatus(SupermavenTaskUpdateMessage { task: "Initializing", status: Complete, percent_complete: None })
2024-11-12T18:49:34.8698189+03:00 [WARN] unhandled message: ActiveRepo(SupermavenActiveRepoMessage { repo_simple_name: None })
2024-11-12T18:49:52.1514548+03:00 [WARN] skipping diagnostics update, no worktree found for path "c:\\Users\\taima\\.rustup\\toolchains\\1.81-x86_64-pc-windows-msvc\\lib\\rustlib\\src\\rust\\library\\std\\src\\path.rs"
2024-11-12T18:49:52.1686617+03:00 [WARN] skipping diagnostics update, no worktree found for path "c:\\Users\\taima\\.cargo\\registry\\src\\index.crates.io-6f17d22bba15001f\\anyhow-1.0.89\\src\\macros.rs"
2024-11-12T18:49:52.1689762+03:00 [WARN] skipping diagnostics update, no worktree found for path "c:\\Users\\taima\\.rustup\\toolchains\\1.81-x86_64-pc-windows-msvc\\lib\\rustlib\\src\\rust\\library\\core\\src\\option.rs"
2024-11-12T18:49:53.002671+03:00 [ERROR] Error { kind: Generic("Input watch path is neither a file nor a directory."), paths: [] }
2024-11-12T18:49:54.1422428+03:00 [WARN] Generic lsp request to rust-analyzer failed: content modified
2024-11-12T18:49:54.1424139+03:00 [ERROR] inlay hint update task for range failed: Error {
    context: "inlay hint fetch task",
    source: Error {
        context: "inlay hints LSP request",
        source: "content modified",
    },
}
2024-11-12T18:50:05.7510449+03:00 [INFO] building git repository, `.git` path in the worktree: "\\\\?\\G:\\CodeProjects\\RustroverProjects\\other\\zed\\.git"
2024-11-12T18:50:05.7543923+03:00 [INFO] building git repository, `.git` path in the worktree: "\\\\?\\G:\\CodeProjects\\RustroverProjects\\other\\zed\\.git"
2024-11-12T18:50:07.0943128+03:00 [INFO] building git repository, `.git` path in the worktree: "\\\\?\\G:\\CodeProjects\\RustroverProjects\\other\\zed\\.git"
2024-11-12T18:50:08.0673015+03:00 [INFO] building git repository, `.git` path in the worktree: "\\\\?\\G:\\CodeProjects\\RustroverProjects\\other\\zed\\.git"
2024-11-12T18:50:08.0813723+03:00 [INFO] building git repository, `.git` path in the worktree: "\\\\?\\G:\\CodeProjects\\RustroverProjects\\other\\zed\\.git"
2024-11-12T18:50:08.0851621+03:00 [INFO] building git repository, `.git` path in the worktree: "\\\\?\\G:\\CodeProjects\\RustroverProjects\\other\\zed\\.git"
2024-11-12T18:50:08.0959883+03:00 [INFO] building git repository, `.git` path in the worktree: "\\\\?\\G:\\CodeProjects\\RustroverProjects\\other\\zed\\.git"
2024-11-12T18:50:08.0996783+03:00 [INFO] building git repository, `.git` path in the worktree: "\\\\?\\G:\\CodeProjects\\RustroverProjects\\other\\zed\\.git"
2024-11-12T18:50:08.1204446+03:00 [INFO] building git repository, `.git` path in the worktree: "\\\\?\\G:\\CodeProjects\\RustroverProjects\\other\\zed\\.git"
2024-11-12T18:50:11.8999929+03:00 [INFO] Creating texture 0x12d3ab345e0 of size 1024x1024x1 and format R8Unorm, name 'atlas', handle 36
2024-11-12T18:52:27.914158+03:00 [ERROR] Failed to send events: HTTP 500
2024-11-12T18:53:56.14188+03:00 [INFO] Using surface present mode FIFO_RELAXED
2024-11-12T18:54:05.9077808+03:00 [INFO] Using surface present mode FIFO_RELAXED

```
<!-- Click above this line and paste or drag-and-drop your log--></details>

--- a/crates/assistant/src/terminal_inline_assistant.rs
+++ b/crates/assistant/src/terminal_inline_assistant.rs
@@ -1073,7 +1073,10 @@ pub enum CodegenEvent {
 
 impl EventEmitter<CodegenEvent> for Codegen {}
 
+#[cfg(not(target_os = "windows"))]
 const CLEAR_INPUT: &str = "\x15";
+#[cfg(target_os = "windows")]
+const CLEAR_INPUT: &str = "\x03";
 const CARRIAGE_RETURN: &str = "\x0d";
 
 struct TerminalTransaction {

--- a/crates/assistant2/src/terminal_codegen.rs
+++ b/crates/assistant2/src/terminal_codegen.rs
@@ -155,7 +155,10 @@ pub enum CodegenEvent {
     Finished,
 }
 
+#[cfg(not(target_os = "windows"))]
 pub const CLEAR_INPUT: &str = "\x15";
+#[cfg(target_os = "windows")]
+pub const CLEAR_INPUT: &str = "\x03";
 const CARRIAGE_RETURN: &str = "\x0d";
 
 struct TerminalTransaction {

